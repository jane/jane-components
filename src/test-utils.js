// @flow

import * as React from 'react'
import renderer from 'react-test-renderer'
import theme from './theme'
import { ThemeProvider } from 'styled-components'

export const create = (C: React.Node): string =>
  renderer.create(<ThemeProvider theme={theme}>{C}</ThemeProvider>).toJSON()
