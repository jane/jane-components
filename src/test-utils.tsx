import * as React from 'react'
import renderer from 'react-test-renderer'
import theme from './theme'
import { ThemeProvider } from 'styled-components'

type RendererJson = {
  type: string
  // eslint-disable-next-line @typescript-eslint/no-explicit-any
  props: { [propName: string]: any }
  children: null | Array<RendererJson>
}

export const create = (C: React.ReactElement): RendererJson | null =>
  renderer.create(<ThemeProvider theme={theme}>{C}</ThemeProvider>).toJSON()
