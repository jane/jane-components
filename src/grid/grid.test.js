// @flow

/* eslint-env jest */

import * as React from 'react'
import { create } from '../test-utils'
import Grid, { GUTTERS, GRID_SIZES } from './grid'

const DIRECTION = ['row', 'row-reverse', 'column', 'column-reverse']
const ALIGN_CONTENT = [
  'stretch',
  'center',
  'flex-start',
  'flex-end',
  'space-between',
  'space-around',
]
const ALIGN_ITEMS = ['flex-start', 'center', 'flex-end', 'stretch', 'baseline']
const WRAP = ['nowrap', 'wrap', 'wrap-reverse']
const JUSTIFY = [
  'flex-start',
  'center',
  'flex-end',
  'space-between',
  'space-around',
  'space-evenly',
]

/* eslint-disable flowtype/require-parameter-type */
describe('grid', (): void => {
  it('renders container', (): void => {
    expect(create(<Grid container />)).toMatchSnapshot()
  })

  it('handles gutters', (): void => {
    GUTTERS.forEach(
      (g): void => {
        expect(create(<Grid spacing={g} />)).toMatchSnapshot()
      }
    )
  })

  it('handles zeroMinWidth', (): void => {
    expect(create(<Grid zeroMinWidth />)).toMatchSnapshot()
  })

  it('handles justify', (): void => {
    JUSTIFY.forEach(
      (j): void => {
        expect(create(<Grid justify={j} />)).toMatchSnapshot()
      }
    )
  })

  it('handles item', (): void => {
    expect(
      <Grid container>
        <Grid item />
      </Grid>
    ).toMatchSnapshot()
  })

  it('handles direction', (): void => {
    DIRECTION.forEach(
      (d): void => {
        expect(create(<Grid direction={d} />)).toMatchSnapshot()
      }
    )
  })

  it('handles alignItems', (): void => {
    ALIGN_ITEMS.forEach(
      (a): void => {
        expect(create(<Grid alignItems={a} />)).toMatchSnapshot()
      }
    )
  })

  it('handles alignContent', (): void => {
    ALIGN_CONTENT.forEach(
      (a): void => {
        expect(create(<Grid alignContent={a} />)).toMatchSnapshot()
      }
    )
  })

  it('handles wrap', (): void => {
    WRAP.forEach(
      (w): void => {
        expect(create(<Grid wrap={w} />)).toMatchSnapshot()
      }
    )
  })

  it('handles grid sizes', (): void => {
    GRID_SIZES.forEach(
      (s): void => {
        expect(create(<Grid xl={s} />)).toMatchSnapshot()
        expect(create(<Grid lg={s} />)).toMatchSnapshot()
        expect(create(<Grid md={s} />)).toMatchSnapshot()
        expect(create(<Grid sm={s} />)).toMatchSnapshot()
        expect(create(<Grid xs={s} />)).toMatchSnapshot()
      }
    )
  })
})
