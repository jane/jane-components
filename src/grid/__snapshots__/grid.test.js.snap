// Jest Snapshot v1, https://goo.gl/fbAQLP

exports[`grid renders 1`] = `
.c0 {
  -webkit-flex: '0 1 auto';
  -ms-flex: '0 1 auto';
  flex: '0 1 auto';
  display: -webkit-box;
  display: -webkit-flex;
  display: -ms-flexbox;
  display: flex;
  -webkit-flex-direction: row;
  -ms-flex-direction: row;
  flex-direction: row;
  -webkit-align-items: stretch;
  -webkit-box-align: stretch;
  -ms-flex-align: stretch;
  align-items: stretch;
  -webkit-align-content: stretch;
  -ms-flex-line-pack: stretch;
  align-content: stretch;
  -webkit-box-pack: start;
  -webkit-justify-content: flex-start;
  -ms-flex-pack: start;
  justify-content: flex-start;
}

<div
  className="c0"
/>
`;
